addons/im_livechat/models/chatbot_script_step.py

```diff
@@ -0,0 +1,199 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import _, api, models, fields
+from odoo.exceptions import ValidationError
+from odoo.fields import Command
+from odoo.osv import expression
+from odoo.tools import html2plaintext, single_email_re, email_normalize
+
+from collections import defaultdict
+
+
+class ChatbotScriptStep(models.Model):
+    _name = 'im_livechat.chatbot.script_step'
+    _description = 'Chatbot Script Step'
+    _order = 'sequence, id'
+
+    message = fields.Text(string='Message', required=True)
+    sequence = fields.Integer(string='Sequence')
```

Comment:
As it is important (some code based on sequence), I would indeed ensure it has a value and maybe even give a default value depending on already-existing steps when possible. Like: when adding a new step, if you already have steps, set sequence as sequence + 1 to try to avoid side effects linked to having the same sequence.

Reply:
We tried a default value but it's not handy because in the default function we do not have access to the "chatbot_id" of this step, meaning we can't do "highest sequence of THIS chatbot +1".

Instead I have opted for a "create" override which will nicely take care of the sequences if not defined.
This allows to batch the sequence handling and seems even more robust than a default value.

Added some tests along the override to ensure the behavior.

Reply:
Indeed, seems better :)